tfhe/src/core_crypto/entities/lwe_packing_keyswitch_key.rs

```diff
@@ -0,0 +1,394 @@
+    /// This docstring exhibits [`LwePackingKeyswitchKey`] primitives usage.
+    ///
+    /// ```
+    /// use tfhe::core_crypto::prelude::*;
+    ///
+    /// // DISCLAIMER: these toy example parameters are not guaranteed to be secure or yield correct
+    /// // computations
+    /// // Define parameters for LwePackingKeyswitchKey creation
+    /// let input_lwe_dimension = LweDimension(600);
+    /// let output_glwe_dimension = GlweDimension(1);
+    /// let output_polynomial_size = PolynomialSize(1024);
+    /// let decomp_base_log = DecompositionBaseLog(4);
+    /// let decomp_level_count = DecompositionLevelCount(5);
+    /// let ciphertext_modulus = CiphertextModulus::new_native();
+    ///
+    /// // Create a new LwePackingKeyswitchKey
+    /// let lwe_pksk = LwePackingKeyswitchKey::new(
+    ///     0u64,
+    ///     decomp_base_log,
+    ///     decomp_level_count,
+    ///     input_lwe_dimension,
+    ///     output_glwe_dimension,
+    ///     output_polynomial_size,
+    ///     ciphertext_modulus,
+    /// );
+    ///
+    /// assert_eq!(lwe_pksk.decomposition_base_log(), decomp_base_log);
+    /// assert_eq!(lwe_pksk.decomposition_level_count(), decomp_level_count);
+    /// assert_eq!(lwe_pksk.input_key_lwe_dimension(), input_lwe_dimension);
+    /// assert_eq!(lwe_pksk.output_key_glwe_dimension(), output_glwe_dimension);
+    /// assert_eq!(
+    ///     lwe_pksk.output_key_polynomial_size(),
+    ///     output_polynomial_size
+    /// );
+    /// assert_eq!(
+    ///     lwe_pksk.output_glwe_size(),
+    ///     output_glwe_dimension.to_glwe_size()
+    /// );
+    /// assert_eq!(lwe_pksk.output_polynomial_size(), output_polynomial_size);
+    /// assert_eq!(lwe_pksk.ciphertext_modulus(), ciphertext_modulus);
+    ///
+    /// // Demonstrate how to recover the allocated container
+    /// let underlying_container: Vec<u64> = lwe_pksk.into_container();
+    ///
+    /// // Recreate a secret key using from_container
+    /// let lwe_pksk = LwePackingKeyswitchKey::from_container(
+    ///     underlying_container,
+    ///     decomp_base_log,
+    ///     decomp_level_count,
+    ///     output_glwe_dimension.to_glwe_size(),
+    ///     output_polynomial_size,
+    ///     ciphertext_modulus,
+    /// );
+    ///
+    /// assert_eq!(lwe_pksk.decomposition_base_log(), decomp_base_log);
+    /// assert_eq!(lwe_pksk.decomposition_level_count(), decomp_level_count);
+    /// assert_eq!(lwe_pksk.input_key_lwe_dimension(), input_lwe_dimension);
+    /// assert_eq!(lwe_pksk.output_key_glwe_dimension(), output_glwe_dimension);
+    /// assert_eq!(
+    ///     lwe_pksk.output_key_polynomial_size(),
+    ///     output_polynomial_size
+    /// );
+    /// assert_eq!(
+    ///     lwe_pksk.output_glwe_size(),
+    ///     output_glwe_dimension.to_glwe_size()
+    /// );
+    /// assert_eq!(lwe_pksk.output_polynomial_size(), output_polynomial_size);
+    /// assert_eq!(lwe_pksk.ciphertext_modulus(), ciphertext_modulus);
+    /// ```
```

Comment:
imo yes, it would more discoverable since its meant to be a example on how to use this entity

in the source code, the struct generally appears first so, its doc string would appears first

also in the generated docs, this example would appear sooner 
e.g https://docs.rs/tfhe/0.3.1/tfhe/core_crypto/entities/lwe_keyswitch_key/struct.LweKeyswitchKey.html
where  there is first a crypto explanation, then there is an example attached to the from_container function, but the example is more general than from_container function usage

Reply:
ok will create an issue to rework the format, but not gonna change this one here